src/database.rs

```diff
@@ -63,13 +61,13 @@ impl DatabaseRef for ReadOnlyDB {
                 balance: U256::from_limbs(acc.balance.0),
                 nonce: acc.nonce.as_u64(),
                 code_size: acc.code_size.as_usize(),
-                code_hash: B256::from(acc.code_hash.to_fixed_bytes()),
-                keccak_code_hash: B256::from(acc.keccak_code_hash.to_fixed_bytes()),
+                code_hash: B256::from(acc.keccak_code_hash.to_fixed_bytes()),
+                poseidon_code_hash: B256::from(acc.code_hash.to_fixed_bytes()),
                 // if None, means CodeDB did not include the code, could cause by: EXTCODESIZE
```

Comment:
A side-note: can we improve this inline documentation by being more descriptive?